### PR TITLE
Store round events in replay meta data file.

### PR DESCRIPTION
**Pull request by Simyon264**
## About the PR

This PR adds an event system which stores, well, events in the `replay_final.yml` of the replay zip. This is just a List of events which can be then parsed by external tools to construct a rough timeline of events in the replay.

## Technical details

TBD

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

TBD

## Breaking changes

Might break other tools that expect the `replay_final.yml` to not contain more properties.

### CODE REVIEW

**Comment by Simyon264** [comment]
Some of the events are recorded directly from the Systems and some using events. I am a bit unsure on wether I want events to be recorded only in the GameTicker using Subscriptions or new code within the systems responsible. I would really like some input on that.

File: Content.Server/Explosion/EntitySystems/ExplosionSystem.cs [hunk 1]
```diff
@@ -255,6 +260,40 @@ public float IntensityToRadius(float totalIntensity, float slope, float maxInten
...
+                Intensity = totalIntensity,
+                Slope = slope,
+                MaxTileIntensity = maxTileIntensity,
+                TileBreakScale = tileBreakScale,
+                MaxTileBreak = maxTileBreak,
+                CanCreateVacuum = canCreateVacuum,
+                Type = typeId
+            });
+        }
+
```
Comment (Brandon-Huu) [inline Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:296]: This is awful

Comment (Simyon264) [reply Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:296]: I agree, but it's a draft. I am still working on it.

File: Content.Server/GameTicking/GameTicker.Replays.cs [hunk 2]
```diff
@@ -150,4 +193,208 @@ private ResPath GetAutoReplayPath()
...
+            {
+                RecordReplayEvent(new GenericPlayerEvent()
+                {
+                    Target = (ReplayEventPlayer) targetInfo,
+                    Severity = ReplayEventSeverity.Medium,
+                    EventType = ReplayEventType.MobDied,
+                    Time = _gameTiming.CurTick.Value
+                });
+            }
+        }
```
Comment (Brandon-Huu) [inline Content.Server/GameTicking/GameTicker.Replays.cs:332]: All of this is so bad.
So much duplicated code

File: Content.Server/RoundEnd/RoundEndSystem.cs [hunk 3]
```diff
@@ -155,10 +156,26 @@ public void RequestRoundEnd(TimeSpan countdownTime, EntityUid? requester = null,
...
                 _adminLogger.Add(LogType.ShuttleCalled, LogImpact.High, $"Shuttle called");
+                _gameTicker.RecordReplayEvent(new ShuttleReplayEvent()
+                {
+                    Source = null,
+                    Countdown = (int)countdownTime.TotalSeconds,
+                    Severity = ReplayEventSeverity.High,
+                    EventType = ReplayEventType.EvacCalled,
+                    Time = _gameTiming.CurTick.Value
+                });
             }
```
Comment (Brandon-Huu) [inline Content.Server/RoundEnd/RoundEndSystem.cs:179]: Most of this code can probably be taken out of the if else blcok

**Comment by metalgearsloth** [comment]
undrafty.

**Pull request closed without merging**